fix(cli): flaky AppContainer sticky todo remeasure test

## Summary

`AppContainer.test.tsx` — "does not remeasure footer height for sticky todo status-only updates" is flaky on CI (macOS runner). It fails intermittently with:

```
AssertionError: expected "spy" to be called 1 times, but got 2 times
  ❯ src/ui/AppContainer.test.tsx:2373
```

CI failure: https://github.com/QwenLM/qwen-code/actions/runs/26231823721/job/77194055495

## Root cause analysis

The test asserts that `measureElement` is **not** called again after rerender when only the todo status changes (`pending` → `in_progress`). It captures the absolute call count after initial render, rerenders, then asserts the count is unchanged.

The `useLayoutEffect` in `AppContainer.tsx:2305-2324` has these deps:

```
[buffer, terminalWidth, terminalHeight, btwItem, dialogsVisible, stickyTodosLayoutKey]
```

The test correctly ensures `stickyTodosLayoutKey` is stable (status is excluded by `getStickyTodosLayoutKey`). However, the assertion relies on the absolute `mock.calls.length` count, which is fragile under React 19's Ink test renderer — where the number of `useLayoutEffect` invocations during initial mount can vary between environments (e.g. StrictMode double-invoke, multiple reconciliation passes).

The companion unhandled rejection (`Config was already initialized`) is a separate StrictMode artifact from `config.initialize()` being called twice, but is not the direct cause of the assertion failure.

## Proposed fix

Replace the absolute call-count assertion with `mockClear()` + `not.toHaveBeenCalled()`, which:
- Clearly expresses the intent ("no new calls after rerender")
- Is independent of how many times `measureElement` was called during mount
- Eliminates the flaky behavior regardless of React reconciler pass count

## Follow-up PR

A fix PR will follow shortly.

--- a/packages/cli/src/ui/AppContainer.test.tsx
+++ b/packages/cli/src/ui/AppContainer.test.tsx
@@ -328,12 +328,6 @@ describe('AppContainer State Management', () => {
     // Mock Config
     mockConfig = makeFakeConfig();
 
-    // makeFakeConfig() returns a real Config whose initialize() throws
-    // "Config was already initialized" on the second call. React double-invokes
-    // the mount effect, so the real method rejects and leaks async renders into
-    // later tests — making footer-measurement assertions flaky. Stub it out.
-    vi.spyOn(mockConfig, 'initialize').mockResolvedValue(undefined);
-
     // Mock config's getTargetDir to return consistent workspace directory
     vi.spyOn(mockConfig, 'getTargetDir').mockReturnValue('/test/workspace');
 
@@ -2344,6 +2338,12 @@ describe('AppContainer State Management', () => {
     });
 
     it('does not remeasure footer height for sticky todo status-only updates', async () => {
+      // Scoped stub: makeFakeConfig().initialize() rejects on React's
+      // double-mount, which leaks async renders and destabilizes the
+      // footer-measurement timing this test depends on. Kept per-test so
+      // unrelated tests in this block still exercise the real init gate.
+      vi.spyOn(mockConfig, 'initialize').mockResolvedValue(undefined);
+
       const historyManager = {
         history: makeTodoHistory('pending'),
         addItem: vi.fn(),
@@ -2382,6 +2382,13 @@ describe('AppContainer State Management', () => {
 
       const heightAfterSettle = capturedUIState.availableTerminalHeight;
 
+      // Switch the mock to a different height so any re-measurement triggered
+      // by the status-only rerender below would change controlsHeight (and
+      // therefore availableTerminalHeight). Without this, the production
+      // same-value short-circuit on setControlsHeight makes the equality
+      // assertion pass even when the optimization regresses.
+      mockedMeasureElement.mockReturnValue({ width: 80, height: 10 });
+
       historyManager.history = makeTodoHistory('in_progress');
       await act(async () => {
         view!.rerender(
@@ -2395,8 +2402,8 @@ describe('AppContainer State Management', () => {
       });
 
       // The sticky todo status change (pending → in_progress) must not alter
-      // the computed terminal height. This asserts on the behavioral outcome
-      // rather than the implementation detail of measureElement call count.
+      // the computed terminal height. Combined with the mock-height swap
+      // above, this fails iff the footer was re-measured.
       expect(capturedUIState.availableTerminalHeight).toBe(heightAfterSettle);
     });
   });

--- a/packages/cli/src/ui/AppContainer.test.tsx
+++ b/packages/cli/src/ui/AppContainer.test.tsx
@@ -328,6 +328,12 @@ describe('AppContainer State Management', () => {
     // Mock Config
     mockConfig = makeFakeConfig();
 
+    // makeFakeConfig() returns a real Config whose initialize() throws
+    // "Config was already initialized" on the second call. React double-invokes
+    // the mount effect, so the real method rejects and leaks async renders into
+    // later tests — making footer-measurement assertions flaky. Stub it out.
+    vi.spyOn(mockConfig, 'initialize').mockResolvedValue(undefined);
+
     // Mock config's getTargetDir to return consistent workspace directory
     vi.spyOn(mockConfig, 'getTargetDir').mockReturnValue('/test/workspace');
 
@@ -2358,7 +2364,11 @@ describe('AppContainer State Management', () => {
           initializationResult={mockInitResult}
         />,
       );
-      const callsAfterInitialRender = mockedMeasureElement.mock.calls.length;
+
+      // Reset call tracking after initial render to isolate rerender calls.
+      // This avoids flakiness from React 19's variable mount-time invocation
+      // count (e.g. StrictMode double-invoke, multiple reconciliation passes).
+      mockedMeasureElement.mockClear();
 
       historyManager.history = makeTodoHistory('in_progress');
       view.rerender(
@@ -2370,9 +2380,7 @@ describe('AppContainer State Management', () => {
         />,
       );
 
-      expect(mockedMeasureElement).toHaveBeenCalledTimes(
-        callsAfterInitialRender,
-      );
+      expect(mockedMeasureElement).not.toHaveBeenCalled();
     });
   });
 

--- a/packages/cli/src/ui/AppContainer.test.tsx
+++ b/packages/cli/src/ui/AppContainer.test.tsx
@@ -14,6 +14,7 @@ import {
   type Mock,
 } from 'vitest';
 import { render, cleanup } from 'ink-testing-library';
+import { useContext, act } from 'react';
 import {
   AppContainer,
   dedupeNewestFirst,
@@ -43,7 +44,6 @@ import {
   type HistoryItemWithoutId,
   ToolCallStatus,
 } from './types.js';
-import { useContext } from 'react';
 import { Box, measureElement } from 'ink';
 
 // Mock useStdout to capture terminal title writes
@@ -2343,7 +2343,7 @@ describe('AppContainer State Management', () => {
       expect(lastCall[2]).toBe(1);
     });
 
-    it('does not remeasure footer height for sticky todo status-only updates', () => {
+    it('does not remeasure footer height for sticky todo status-only updates', async () => {
       const historyManager = {
         history: makeTodoHistory('pending'),
         addItem: vi.fn(),
@@ -2356,31 +2356,48 @@ describe('AppContainer State Management', () => {
       mockedUseTerminalSize.mockReturnValue({ columns: 80, rows: 24 });
       mockedMeasureElement.mockReturnValue({ width: 80, height: 4 });
 
-      const view = render(
-        <AppContainer
-          config={mockConfig}
-          settings={mockSettings}
-          version="1.0.0"
-          initializationResult={mockInitResult}
-        />,
-      );
+      let view: ReturnType<typeof render>;
+      await act(async () => {
+        view = render(
+          <AppContainer
+            config={mockConfig}
+            settings={mockSettings}
+            version="1.0.0"
+            initializationResult={mockInitResult}
+          />,
+        );
+      });
 
-      // Reset call tracking after initial render to isolate rerender calls.
-      // This avoids flakiness from React 19's variable mount-time invocation
-      // count (e.g. StrictMode double-invoke, multiple reconciliation passes).
-      mockedMeasureElement.mockClear();
+      // Let any pending state updates from useLayoutEffect settle.
+      await act(async () => {
+        view!.rerender(
+          <AppContainer
+            config={mockConfig}
+            settings={mockSettings}
+            version="1.0.0"
+            initializationResult={mockInitResult}
+          />,
+        );
+      });
+
+      const heightAfterSettle = capturedUIState.availableTerminalHeight;
 
       historyManager.history = makeTodoHistory('in_progress');
-      view.rerender(
-        <AppContainer
-          config={mockConfig}
-          settings={mockSettings}
-          version="1.0.0"
-          initializationResult={mockInitResult}
-        />,
-      );
+      await act(async () => {
+        view!.rerender(
+          <AppContainer
+            config={mockConfig}
+            settings={mockSettings}
+            version="1.0.0"
+            initializationResult={mockInitResult}
+          />,
+        );
+      });
 
-      expect(mockedMeasureElement).not.toHaveBeenCalled();
+      // The sticky todo status change (pending → in_progress) must not alter
+      // the computed terminal height. This asserts on the behavioral outcome
+      // rather than the implementation detail of measureElement call count.
+      expect(capturedUIState.availableTerminalHeight).toBe(heightAfterSettle);
     });
   });
 

